### DISCUSSION TITLE
I'm confused about how to edit layouts.

### DISCUSSION THREAD

**Discussion by zMenta** [zellij-org/zellij#2445] [Q&A] [answered]
For my use case I just wanted a the tab function for zellij, since I just discovered this tool and I did never used tmux or other similar tools. So I just wanted to have a clean zellij interface.

With the default layout `default_layout  "default" `, I like that the tab bar is shown on the top of the screen, like so: 

![image](https://github.com/zellij-org/zellij/assets/70714721/31e2e93e-9e70-4386-bb08-71cf36669d0e)

But I like the feel of the `default_layout  "compact" `, but wanted the tab bar on the top of the screen instead of the bottom.

![image](https://github.com/zellij-org/zellij/assets/70714721/49cbe3a1-6f86-4c40-800e-bded382f5e8d)

How can I achieve this?

**Comment by Lockszmith-GH** [comment]
Did you review the (excellent) documentation about [Creating a Layout](https://zellij.dev/documentation/creating-a-layout.html#creating-a-layout)?

The first thing it shows you is how to actually see the default layout:
```
$ zellij setup --dump-layout default > /tmp/my-quickstart-layout-file.kdl
```

It also has [this section](https://zellij.dev/documentation/creating-a-layout.html#default-tab-template)

**Reply by Lockszmith-GH** [reply]
Currently, the output of:
```
zellij setup --dump-layout default
```
is:
```
layout {
    pane size=1 borderless=true {
        plugin location="zellij:tab-bar"
    }
    pane
    pane size=2 borderless=true {
        plugin location="zellij:status-bar"
    }
}
```

**Reply by zMenta** [reply]
Yes, but I'm still finding a bit confusing how to do it. But after trying to mess around again because of your reply, it made a little more sense to me.

Correct if I'm wrong. I understood that to create a layout you need to create a layouts directory containing layouts. Just like the default layout you replied.

So inside, zellij .config directory I created a layouts directory. Create a `compact-top.kdl` file inside it.

On the zellij config file I added `layout_dir "~/.config/zellij/layouts"` and `default_layout "compact-top"`

But when trying to load the layout I receive the following error:
`IoError: The layout was not found, File: compact-top`

**Reply by zMenta** [reply, marked as answer]
Reading further the documentation I found out that zellij searches for the directory /config/layouts by default. So just needed to create the layouts directory without changing the _default_layout_ variable.

Just created my layout under the layouts directory and used the file name in the config file. Now it works as I expected :)

![image](https://github.com/zellij-org/zellij/assets/70714721/87a33146-0ae4-486e-b8da-49c38fbb2ffa)

**Comment by OdysseusOperator-eviden** [comment]
A quick look into the source code revealed that you can configure the layout dir.

I add following line to my zellij config 
layout_dir "/home/<user>/.config/zellij/layouts"

`mkdir /home/<user>/.config/zeelij/layouts`
And then i copied my layout into this folder.

Zellij does not expand ~ maybe $Home work, but just using the full path was easier in this case.
